gitlab/mixins.py

```diff
@@ -173,7 +172,11 @@ def refresh(self, **kwargs: Any) -> None:
             GitlabGetError: If the server cannot perform the request
         """
         if self._id_attr:
-            path = f"{self.manager.path}/{self.id}"
+            if self._lazy:
+                id = self.get_id()
+            else:
+                id = self.encoded_id
+            path = f"{self.manager.path}/{id}"
```

Comment:
Unfortunately I don't think the logic needs to be here :/

The way we've done this, maybe because `encoded_id`, it's basically for anything that happens after a lazy get, I think.

```
>>> gl = gitlab.Gitlab(private_token=pat)
>>> project = gl.projects.get("nejc/private-project", lazy=True)
>>> project.description = "My stuff"
>>> project.save()
Traceback (most recent call last):
...
```

Maybe we just don't need to url-encode for lazy objects at all with this fix, if we just ensure it's done for all real operations. It will also give prettier `my_object.id` for lazy objects.

Reply:
Okay. I believe I have fixed things by adding the `EncodedId` string class.